airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -386,10 +386,14 @@ def execute(self, context: 'Context'):
                 result = self.extract_xcom(pod=self.pod)
             remote_pod = self.pod_manager.await_pod_completion(self.pod)
         finally:
-            self.cleanup(
-                pod=self.pod or self.pod_request_obj,
-                remote_pod=remote_pod,
-            )
+            pod = self.pod or self.pod_request_obj
+            if self.find_pod(pod.metadata.namespace, context=context):
```

Comment:
I wonder if we should move this into `cleanup` instead. If we are going to grab the pod from the k8s api, we might as well use that status in `cleanup` as it could be more up to date.